iptables/table_test.go

```diff
@@ -482,6 +484,59 @@ func describeEmptyDataplaneTests(dataplaneMode string) {
 			})
 		})
 	})
+
+	Describe("applying updates when underlying iptables have changed", func() {
+		BeforeEach(func() {
+			table.SetRuleInsertions("FORWARD", []Rule{
+				{Action: AcceptAction{}},
+				{Action: DropAction{}},
+			})
+			table.UpdateChains([]*Chain{
+				{Name: "cali-foobar", Rules: []Rule{
+					{Action: AcceptAction{}},
+					{Action: DropAction{}},
+				}},
+			})
+			table.Apply()
+		})
+		It("should be in the dataplane", func() {
+			Expect(dataplane.Chains).To(Equal(map[string][]string{
+				"FORWARD": {
+					"-m comment --comment \"cali:3gUkOfVeYRgMeHF4\" --jump ACCEPT",
+					"-m comment --comment \"cali:8MgbRleZ5Rc5cBEf\" --jump DROP",
+				},
+				"INPUT":  {},
+				"OUTPUT": {},
+				"cali-foobar": {
+					"-m comment --comment \"cali:42h7Q64_2XDzpwKe\" --jump ACCEPT",
+					"-m comment --comment \"cali:0sUFHicPNNqNyNx8\" --jump DROP",
+				},
+			}))
+		})
+		Describe("then truncating the chain, with the iptables changed before iptables-restore", func() {
+			BeforeEach(func() {
+				dataplane.InsertRandomRuleInForward = true
+				table.SetRuleInsertions("FORWARD", []Rule{
+					{Action: DropAction{}},
```

Comment:
Suggest you throw a comment onto each rule you're using in these tests.  It'd be easy to have bugs where one of the old rules hung around but the new drop rule you asked for got lost.